### ISSUE TITLE
int32 vs long, and filters.blacklist

### ISSUE THREAD

**Issue by konstan** [DarioBalinzo/kafka-connect-elasticsearch-source#38]
Hello @DarioBalinzo,

After some time, the connector stops working for me with the following error.

```
[2021-02-28 21:36:03,823] INFO fetching from nuvla-nuvlabox-status (com.github.dariobalinzo.task.ElasticSourceTask:166)
[2021-02-28 21:36:03,823] INFO found last value 2021-02-28T21:35:46.489Z (com.github.dariobalinzo.task.ElasticSourceTask:168)
[2021-02-28 21:36:03,827] ERROR error (com.github.dariobalinzo.task.ElasticSourceTask:180)
org.apache.kafka.connect.errors.DataException: Invalid Java object for schema type INT32: class java.lang.Long for field: "bytes-transmitted"
        at org.apache.kafka.connect.data.ConnectSchema.validateValue(ConnectSchema.java:245)
        at org.apache.kafka.connect.data.Struct.put(Struct.java:216)
        at org.apache.kafka.connect.data.Struct.put(Struct.java:203)
        at com.github.dariobalinzo.schema.StructConverter.convertDocumentStruct(StructConverter.java:47)
        at com.github.dariobalinzo.schema.StructConverter.convertListOfObject(StructConverter.java:114)
        at com.github.dariobalinzo.schema.StructConverter.lambda$convertListToAvroArray$0(StructConverter.java:81)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
        at com.github.dariobalinzo.schema.StructConverter.convertListToAvroArray(StructConverter.java:82)
        at com.github.dariobalinzo.schema.StructConverter.convertDocumentStruct(StructConverter.java:49)
        at com.github.dariobalinzo.schema.StructConverter.covertMapToAvroStruct(StructConverter.java:96)
        at com.github.dariobalinzo.schema.StructConverter.convertDocumentStruct(StructConverter.java:51)
        at com.github.dariobalinzo.schema.StructConverter.convert(StructConverter.java:37)
        at com.github.dariobalinzo.task.ElasticSourceTask.parseResult(ElasticSourceTask.java:214)
        at com.github.dariobalinzo.task.ElasticSourceTask.poll(ElasticSourceTask.java:170)
        at org.apache.kafka.connect.runtime.WorkerSourceTask.poll(WorkerSourceTask.java:289)
        at org.apache.kafka.connect.runtime.WorkerSourceTask.execute(WorkerSourceTask.java:256)
        at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:185)
        at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:234)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
```

The `bytes-transmitted` value is always growing. And eventually, the connector chocks (`2313987073` below) as java int32 max value is 2,147,483,647. 

```
    "net-stats" : [ {
      "interface" : "docker_gwbridge",
      "bytes-transmitted" : 0,
      "bytes-received" : 0
    }, {
      "interface" : "eth0",
      "bytes-transmitted" : 2313987073,
      "bytes-received" : 139486370
    }, {
      "interface" : "veth3a85f64",
      "bytes-transmitted" : 37243658,
      "bytes-received" : 20884091
    }, {
      "interface" : "veth23d5664",
      "bytes-transmitted" : 73770,
      "bytes-received" : 7149
    }, {
      "interface" : "veth480948a",
      "bytes-transmitted" : 1702,
      "bytes-received" : 0
    }, {
      "interface" : "veth4bdbfca",
      "bytes-transmitted" : 47246,
      "bytes-received" : 68340
    }, {
      "interface" : "vethdd78a5f",
      "bytes-transmitted" : 1772,
      "bytes-received" : 0
    }, {
      "interface" : "veth9ba3f40",
      "bytes-transmitted" : 350,
      "bytes-received" : 0
    }, {
      "interface" : "docker0",
      "bytes-transmitted" : 0,
      "bytes-received" : 0
    }, {
      "interface" : "vethc58514b",
      "bytes-transmitted" : 2128,
      "bytes-received" : 0
    }, {
      "interface" : "br-4788132ed466",
      "bytes-transmitted" : 35697531,
      "bytes-received" : 15309643
    }, {
      "interface" : "veth9ab8cd6",
      "bytes-transmitted" : 9663155,
      "bytes-received" : 2041434
    }, {
      "interface" : "veth5fb3fb4",
      "bytes-transmitted" : 6890675,
      "bytes-received" : 11141372
    }, {
      "interface" : "lo",
      "bytes-transmitted" : 18483607,
      "bytes-received" : 18483607
    } ]
```

I think the conversion should treat all int values as longs. Or alternatively, provide a way to define this through a schema?

At the moment I'm not using `net-stats` at all. So, as the workaround, I will now try to `filters.json_cast: net-stats` (hoping it will not parse it at all). 

At the same time, you have `filters.whitelist` parameter. Is it possible to have `filters.blacklist`?

Thank you.

**Comment by konstan** [comment]
Just to let you know that this workaround worked for me: `"filters.json_cast": "resources.net-stats"`

**Comment by konstan** [comment]
... but I want to start using `net-stats` in the very near future.

**Comment by DarioBalinzo** [comment]
Sure,
to avoid the integer overflow I can always treat integer as long. 

I will try to publish the next release in few days, including this fix among the blacklist filter.

**Comment by DarioBalinzo** [comment]
A configuration to choose the integer precision might be probably too difficult to understand and use for many users, and I would like that the connector "just" works without particular configuration.

**Comment by DarioBalinzo** [comment]
I've published a new 1.4 pre relase version, related to the open issues. The 1.4.1 branch contains the updated documentation.
The master branch still refer to version 1.3, the version published on the official confluent hub. After some time I will check with the confluent team how to update also their version in the hub to the final 1.4 version.
